### PR TITLE
Turn on nullability warnings and clean up others

### PR DESCRIPTION
**Pull request by AdamNorberg**
The nullability detector didn't see any problems with our existing logic (it will complain more when actual nullability annotations are required) but in looking for them I cleaned up a bunch of other warnings.

The code is still not clean of warnings, but I'm not confident what to do about the other ones, or it's disruptive (changing namespace or rearranging the project), or it looks like it's a warning due to incomplete stub code that should be left in its incomplete state.

### CODE REVIEW

File: engine/calculus/LinearSpline1D.cs [hunk 1]
```diff
@@ -68,17 +66,17 @@ public LinearSpline1D(SortedList<double, double> points)
 			DebugUtil.AssertAllFinite(Points, nameof(Points));
 
 			// Calculate the coefficients for each segment of the spline:
-			_parameters = new double[points.Count];
+			var parameters = new double[points.Count];
```
Comment (Lathreas) [inline engine/calculus/LinearSpline1D.cs:69]: Oop, I see now that the entire calculation of 'parameters' can probably be removed anyway. It appears to be calculated again during GetNthDerivativeAt - probably a flaw that slipped in during the conversion from Cubic/Quadratic (which need complicated parameters) to Linear (whose parameters are basically trivial). Feel free to remove any reference to 'parameters' within LinearSpline1D as part of the cleanup; they're not referenced elsewhere anyway. (Though that could be done in a different PR)

File: engine/geometry/Hemisphere.cs [hunk 2]
```diff
@@ -178,7 +178,7 @@ public override List<Vertex> GenerateVertexList(int resolutionU, int resolutionV
 			List<Vertex> output = new List<Vertex>(CalculateVertexCount(resolutionU, resolutionV));
 			
 			// Load all required variables:
-			dvec3 up = dvec3.UnitZ;
+			// dvec3 up = dvec3.UnitZ;
```
Comment (Lathreas) [inline engine/geometry/Hemisphere.cs:181]: Ah, this one is also redundant indeed. Feel free to remove entirely!

**Review by Lathreas — COMMENTED** [review]
Overall the refactorings look very good! I have one question about the documentation comments, but if that's no concern, then feel free to merge.

File: engine/calculus/ContinuousMap.cs [hunk 3]
```diff
@@ -20,8 +20,8 @@ namespace FreedomOfFormFoundation.AnatomyEngine.Calculus
 {
 	/// A ContinuousMap is an abstract class that defines some function that outputs a value of type `TOut` for each
 	/// input of type `TIn`. Mathematically, this means it defines a mapping from one space to another. For example, a
-	/// ContinuousMap<double, double> defines a function that takes a double value and outputs a double value, such as
-	/// a polynomial function. ContinuousMap<Vector2, double> takes a Vector2 as input and gives a double as output, which
+	/// ContinuousMap{double, double} defines a function that takes a double value and outputs a double value, such as
+	/// a polynomial function. ContinuousMap{Vector2, double} takes a Vector2 as input and gives a double as output, which
```
[the remark this answers is not in the document]
Comment (AdamNorberg) [reply engine/calculus/ContinuousMap.cs:24]: This is Microsoft's official way to write generic types in XML-format comments. Check out the bullet point in the intro about `<see cref="...">` in https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#introduction - and though they're not explicit about it, they expect this formatting everywhere.

File: engine/calculus/LinearSpline1D.cs (hunk 1 above)
Comment (AdamNorberg) [reply engine/calculus/LinearSpline1D.cs:69]: removed.

File: engine/geometry/Hemisphere.cs (hunk 2 above)
Comment (AdamNorberg) [reply engine/geometry/Hemisphere.cs:181]: removed.

**Review by Lathreas — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**